crates/red_knot_python_semantic/src/semantic_index/symbol.rs

```diff
@@ -387,6 +403,12 @@ impl NodeWithScopeRef<'_> {
             NodeWithScopeRef::ClassTypeParameters(class) => {
                 NodeWithScopeKey::ClassTypeParameters(NodeKey::from_node(class))
             }
+            NodeWithScopeRef::TypeAlias(type_alias) => {
+                NodeWithScopeKey::TypeAliasTypeParameters(NodeKey::from_node(type_alias))
```

Comment:
This should be `NodeWithScopeKey::TypeAlias`, not `NodeWithScopeKey::TypeAliasTypeParameters`. (Will also require adding the `NodeWithScopeKey::TypeAlias` variant below.) Otherwise, both the type alias' type params scope and its inner scope will try to use the same key in the scopes-by-node mapping.

It apparently works anyway because we never actually try to look up the type parameters scope, and we insert the inner scope into the map second, so it replaces the type-params scope, and we always end up getting the inner scope back from a lookup.

We should add a debug assertion in `SemanticIndexBuilder::push_scope_with_parent` that our `self.scopes_by_node.insert(...)` call always returns `None` (that is, it never stomps on an existing scope recorded for the same node key.)